go/ql/lib/semmle/go/Expr.qll

```diff
@@ -980,18 +980,36 @@ class StructTypeExpr extends @structtypeexpr, TypeExpr, FieldParent {
  * Examples:
  *
  * ```go
- * func(a, b int, c float32) (float32, bool)
+ * func(a int, b, c float32) (float32, bool)
  * ```
  */
 class FuncTypeExpr extends @functypeexpr, TypeExpr, ScopeNode, FieldParent {
   /** Gets the `i`th parameter of this function type (0-based). */
   ParameterDecl getParameterDecl(int i) { result = this.getField(i) and i >= 0 }
 
-  /** Gets a parameter of this function type. */
+  /**
+   * Gets a parameter declaration of this function type.
+   *
+   * For example, for `func(a int, b, c float32) (float32, bool)` the result is
+   * `a int` or `b, c float32`.
+   */
   ParameterDecl getAParameterDecl() { result = this.getParameterDecl(_) }
 
-  /** Gets the number of parameters of this function type. */
-  int getNumParameter() { result = count(this.getAParameterDecl()) }
+  /**
+   * Gets the number of parameter declarations of this function type.
+   *
+   * For example, for `func(a int, b, c float32) (float32, bool)` the result is 2:
+   * `a int` and `b, c float32`.
+   */
+  int getNumParameterDecl() { result = count(this.getAParameterDecl()) }
+
+  /**
+   * Gets the number of parameters of this function type.
+   *
+   * For example, for `func(a int, b, c float32) (float32, bool)` the result is 3:
+   * `a`, `b` and `c`.
+   */
+  int getNumParameter() { result = count(this.getAParameterDecl().getANameExpr()) }
```

Comment:
Question to sanity check: this is a breaking change to `getNumParameter`. While it doesn't look like any of our own libraries are affected by this change, other than the changes you made below, and it was arguably a bug, do we know if anyone else might be impacted by this? Would it be safer to keep the existing definition for `getNumParameter` and instead come up with a new name for the new definition?

Reply:
It is a breaking change. But anyone who used it almost certainly wanted the behaviour of the new one. They do the same thing except in the rare case that someone uses a parameter declaration more than one parameter, like `a, b int`. In the change note I chose "minorAnalysis", which means "An API has changed in a way that may affect the results produced by a query that consumes the API, but only in scenarios that affect relatively few users." (cf [the guide on making change notes](https://github.com/github/codeql/blob/main/docs/change-notes.md).

Reply:
Agreed, normally we would avoid this but in this case it's likely a bugfix more often than a breaking change... and the number of users in each category are likely to be zero

Reply:
Another argument in favour of this change is that there are member predicates called `getNumParameter` on other classes (`Function` and `SignatureType`) which do what you'd expect, so it makes sense that this one should do the same.